CMakeLists.txt

```diff
@@ -0,0 +1,45 @@
+cmake_minimum_required(VERSION 3.20)
+project(libCEED LANGUAGES C CXX)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_EXPORT_COMPILE_COMMANDS 1)
+
+if (CEED_ENABLE_CUDA)
+  enable_language(CUDA)
+endif()
+
+file(GLOB_RECURSE interface_c_sources ${PROJECT_SOURCE_DIR}/interface/*.c)
+file(GLOB_RECURSE interface_cpp_sources ${PROJECT_SOURCE_DIR}/interface/*.cpp)
+
+if (NOT CEED_ENABLE_HIP)
+    list(REMOVE_ITEM interface_c_sources ${PROJECT_SOURCE_DIR}/interface/ceed-hip.c)
+endif()
+
+if (NOT CEED_ENABLE_CUDA)
+    list(REMOVE_ITEM interface_c_sources ${PROJECT_SOURCE_DIR}/interface/ceed-cuda.c)
+endif()
+
+file(GLOB_RECURSE gallery_sources ${PROJECT_SOURCE_DIR}/gallery/*.c)
+
+add_library(CEED ${interface_c_sources} ${interface_cpp_sources} ${gallery_sources})
+add_subdirectory(backends)
+target_include_directories(CEED PUBLIC include)
+target_compile_definitions(CEED PUBLIC CEED_JIT_SOURCE_ROOT_DEFAULT="${PROJECT_SOURCE_DIR}/include")
+
+# TODO: which flags should be enabled for a given platform?
+target_compile_options(CEED PRIVATE "-march=native" 
+                                    "-fvectorize" 
+                                    "-fslp-vectorize" 
+                                    "-ffp-contract=fast")
+
+if (CEED_ENABLE_TESTING)
+  add_subdirectory(tests)
+endif()
```

Comment:
Looks like this needs `cmake -DCEED_ENABLE_TESTING=1`. I get link errors for lack of libm.
```
/usr/bin/ld/usr/bin/ld: : /usr/bin/ld: ../libCEED.a(ceed-basis.c.o)../libCEED.a(ceed-basis.c.o): in function `: in function `CeedQRFactorization':
ceed-basis.c:../libCEED.a(ceed-basis.c.o)(: in function `.textCeedQRFactorization+0xCeedQRFactorization':
1afc':
ceed-basis.c:)ceed-basis.c:(: undefined reference to `(.textsqrt.text+0x'
+0x1afc/usr/bin/ld1afc): ): undefined reference to `: undefined reference to `../libCEED.a(ceed-basis.c.o)sqrtsqrt: in function `'
```
It also looks like `make -j` is producing jumbled output, not sure why.

Reply:
oops, I rarely use C directly so I didn't realize you had to explicitly link a separate library to use `sqrt`! I don't know how it managed to compile successfully on my mac, perhaps apple's clang variant links in libm implicitly.